lib/screens/rootscreen.dart

```diff
@@ -30,170 +30,169 @@ class _RootScreenState extends State<RootScreen> {
-                  size: 20.0,
-                  color: Theme.of(context).accentColor,
-                ),
-              ],
-              animationDuration: Duration(milliseconds: 200),
-              index: 0,
-              animationCurve: Curves.bounceInOut,
-              onTap: (index) {
+      appBar: AppBar(
+        title: Text(_appBarTitle == '' ? 'Dashboard' : _appBarTitle),
+        actions: isHome
+            ? <Widget>[
+          IconButton(
+              icon: Icon(Icons.filter_list),
+              iconSize: 30.0,
+              //color: Theme.of(context).accentColor,
+              color: Colors.black,
+              onPressed: () {
+                return Navigator.push(context,
+                    MaterialPageRoute(builder: (context) {
+                      return Filter();
+                    }));
+              }),
+          IconButton(
+              icon: Icon(Icons.settings),
+              //color: Theme.of(context).accentColor,
+              color: Colors.black,
+              onPressed: () {
+                return Navigator.push(context,
+                    MaterialPageRoute(builder: (context) {
+                      return Settings();
+                    }));
+              }),
+          FlatButton.icon(
+            icon: Icon(Icons.person),
+            onPressed: () async {
+              setState(() => loading = true);
+              try {
+                await _auth.signOut();
+                setState(() => loading = false);
+              } catch (e) {
                 setState(() {
-                  justLoggedin = false;
-                  switch (index) {
-                    case 0:
-                      choose = Dashboard();
-                      _appBarTitle = 'Dashboard';
-                      isHome = true;
-                      break;
-                    case 1:
-                      choose = MyRequests();
-                      _appBarTitle = 'My Requests';
-                      isHome = false;
-                      break;
-                    case 2:
-                      choose = Messages();
-                      _appBarTitle = 'Messages';
-                      isHome = false;
-                      break;
-                    case 3:
-                      choose = Notifications();
-                      _appBarTitle = 'Notifications';
-                      isHome = false;
-                      break;
-                    case 4:
-                      choose = MyProfile();
-                      _appBarTitle = 'My Profile';
-                      isHome = false;
-                      break;
-                    // default:
-                    //   0;
-                    // choose = Dashboard();
-                    // _appBarTitle = 'Share A Cab';
-                    // isHome = true;
-                  }
+                  error = e.message;
+                  setState(() => loading = false);
                 });
-              },
-            ),
```

Comment:
Move this appBar to the respective pages. Instead of a common one, you need different appBar for different tabs. Maybe this is the reason you were having difficulty in implementing this https://www.youtube.com/watch?v=FPcl1tu0gDs&t=5s. After that it will be easy.
Maybe logout will not work after that, but don't worry about that. Logout will be fixed by kshitij later if it does not work.
Simply make different appBar in different screens right now